src/header/header.scss

```diff
@@ -291,6 +267,55 @@
   }
 }
 
+@mixin iui-header-button-active {
```

Comment:
@mayank99 Is this good enough? This doesn't break any other button styles and works for split button.

Reply:
Can you move this mixin below split button? I can't even see the diff properly right now

Reply:
Are you sure we need all these extra rules? I thought we would need fewer overrides after using the mixin.

Reply:
Yeah, this is better. I did notice two different problems:

1. Focus styling for left button looks wrong (it should extend all the way to the left edge).
![image](https://user-images.githubusercontent.com/9084735/129087703-532ce7db-7d50-4ebc-8647-65226b86f1f9.png)

2. Disabled buttons still have hover effect and blue stripe.
![image](https://user-images.githubusercontent.com/9084735/129088011-584a87fc-77e9-4db3-bf5e-cbf19e8554bf.png)